pkg/controller/clusters/workers.go

```diff
@@ -0,0 +1,122 @@
+package clusters
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	kubeadmv1 "sigs.k8s.io/cluster-api/bootstrap/kubeadm/api/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	anywherev1 "github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+	"github.com/aws/eks-anywhere/pkg/clusterapi"
+	"github.com/aws/eks-anywhere/pkg/collection"
+	"github.com/aws/eks-anywhere/pkg/controller"
+	"github.com/aws/eks-anywhere/pkg/controller/serverside"
+)
+
+// Workers represents the CAPI spec for an eks-a cluster's workers.
+type Workers struct {
+	Groups []WorkerGroup
+}
+
+// objects returns a list of API objects for a collection of worker groups.
+func (w *Workers) objects() []client.Object {
+	objs := make([]client.Object, 0, len(w.Groups)*3)
+	for _, g := range w.Groups {
+		objs = append(objs, g.objects()...)
+	}
+
+	return objs
+}
+
+// WorkerGroup represents the CAPI spec for an eks-a worker group.
+type WorkerGroup struct {
+	KubeadmConfigTemplate   *kubeadmv1.KubeadmConfigTemplate
+	MachineDeployment       *clusterv1.MachineDeployment
+	ProviderMachineTemplate client.Object
+}
+
+func (g *WorkerGroup) objects() []client.Object {
+	return []client.Object{
+		g.KubeadmConfigTemplate,
+		g.MachineDeployment,
+		g.ProviderMachineTemplate,
+	}
+}
+
+// ToWorkers converts the generic clusterapi Workers definition to the concrete one defined
+// here. It's just a helper for callers generating workers spec using the clusterpai package.
+func ToWorkers[M clusterapi.Object[M]](capiWorkers *clusterapi.Workers[M]) *Workers {
+	w := &Workers{
+		Groups: make([]WorkerGroup, 0, len(capiWorkers.Groups)),
+	}
+
+	for _, g := range capiWorkers.Groups {
+		w.Groups = append(w.Groups, WorkerGroup{
+			MachineDeployment:       g.MachineDeployment,
+			KubeadmConfigTemplate:   g.KubeadmConfigTemplate,
+			ProviderMachineTemplate: g.ProviderMachineTemplate,
+		})
+	}
+
+	return w
+}
+
+// ReconcileWorkersForEKSA orchestrates the worker node reconciliation logic for a particular EKS-A cluster.
```

Comment:
Do you mind explaining why we need reconcile workers for EKSA differently?

Reply:
What do you mean by differently?
It's not different than any one capi cluster, it actually uses the `ReconcileWorkers` implementation. But `ReconcileWorkersForEKSA` takes an eksa cluster as input, which it uses to derive which one is the right capi cluster.

I think I might not be fully getting the question 😓

Reply:
I was essentially asking for the difference between `ReconcileWorkers` and `ReconcileWorkersForEKSA`